ACs/Haier/Haier_AC_AS18GD3HRA.ir

```diff
@@ -0,0 +1,52 @@
+Filetype: IR signals file
+Version: 1
+# 
+# Haier AS18GD3HRA split type Air Conditioner
+# 
+name: ON
+type: raw
+frequency: 38000
+duty_cycle: 0.330000
+data: 3027 3093 3029 4469 541 1675 542 557 542 1675 542 557 543 557 542 1675 542 1675 542 557 542 1674 543 557 542 556 543 556 543 1673 544 557 542 1674 542 557 542 556 543 557 542 557 542 558 541 557 543 557 542 557 542 557 543 555 544 557 542 557 542 557 542 556 543 558 541 557 542 556 544 558 541 1674 542 558 541 557 542 556 543 557 542 557 542 555 544 557 543 557 542 1674 542 556 543 557 542 557 542 557 542 558 541 557 542 557 542 556 543 556 543 556 543 556 543 556 543 557 542 1674 542 557 542 556 543 558 541 557 542 557 542 559 541 559 540 557 542 556 543 558 542 557 542 556 543 557 542 558 541 558 541 556 543 558 541 557 542 557 542 557 542 556 543 557 542 557 542 558 541 557 542 557 542 558 541 557 542 557 542 558 541 558 541 557 542 556 543 557 542 557 542 556 543 558 541 558 541 557 542 559 540 556 543 557 542 557 542 556 543 1676 540 558 541 1673 543 557 542 556 543 557 542 1675 541 557 542 1675 541 558 541 1674 542
+# 
+name: OFF
+type: raw
+frequency: 38000
+duty_cycle: 0.330000
+data: 3026 3097 3026 4471 542 1676 541 557 542 1675 542 559 540 556 543 1676 541 1675 542 558 541 1675 542 556 543 557 542 558 541 557 542 557 542 557 542 558 542 557 542 557 542 557 542 559 540 557 542 557 542 557 542 557 542 558 541 557 542 558 541 558 541 557 542 557 542 557 542 557 542 556 543 557 542 557 542 557 542 556 543 557 542 558 541 557 542 557 542 557 542 1674 542 557 542 556 543 557 542 557 542 558 542 557 542 557 542 557 542 556 543 557 542 557 542 557 542 557 542 1674 542 557 542 556 543 557 542 556 543 557 542 556 543 557 542 557 542 558 541 557 542 556 543 557 542 557 542 556 543 557 542 557 542 557 542 559 540 555 544 557 542 557 542 558 541 556 543 558 541 557 542 557 542 557 542 558 541 557 542 558 541 557 542 557 542 556 543 556 543 558 541 556 543 556 543 558 541 557 543 557 542 557 542 558 541 557 542 556 543 1673 543 557 542 1674 542 1675 541 1674 542 557 542 557 542 1673 543 556 543 1675 541 1674 542
+# 
+name: 25 C
```

Comment:
Each temperature has it's own unique code. Do I need to capture each code individually, or is there a way to somehow keep state internally and just use a +/- magic button?

Reply:
There is no magic button unfortunately. This is a known issue with AC units.